compiler/qsc_eval/src/val.rs

```diff
@@ -16,18 +16,32 @@ pub enum Value {
     Array(Rc<Vec<Value>>),
     BigInt(BigInt),
     Bool(bool),
-    Closure(Rc<[Value]>, StoreItemId, FunctorApp),
+    Closure(Box<Closure>),
     Double(f64),
     Global(StoreItemId, FunctorApp),
     Int(i64),
     Pauli(Pauli),
     Qubit(Qubit),
-    Range(Option<i64>, i64, Option<i64>),
+    Range(Box<Range>),
```

Comment:
Why did these have to become boxed?

Reply:
That was another performance enhancement that was in the branch I was working on, and I didn't want to tease it out. These two tuples were each larger than a pointer, so boxing them makes the whole structure smaller and improves performance. This is one of the recommendations in the Rust Performance Book as a possible optimization: https://nnethercote.github.io/perf-book/heap-allocations.html#box